mlrun/runtimes/nuclio/function.py

```diff
@@ -495,6 +495,14 @@ def add_v3io_stream_trigger(
         extra_attributes = extra_attributes or {}
         if ack_window_size:
             extra_attributes["ackWindowSize"] = ack_window_size
+
+        access_key = kwargs.pop("access_key", None)
+        if access_key:
+            logger.warning(
+                "The access_key parameter is deprecated, use the V3IO_ACCESS_KEY environment variable instead"
+            )
```

Comment:
If access key is not given and nuclio version doesn't support generate then it will still appear in the dict.
What will happen if we remove it from the dict once nuclio created the trigger? will redeploy work?

Reply:
> What will happen if we remove it from the dict once nuclio created the trigger? will redeploy work?

Nope, because it will take the new request body as the "desired" state.
The the access key will be missing and it fail to connect to the v3io stream.